### PR TITLE
[Feature Service] Updates to Javadoc

### LINKED ISSUE #291: [Features] Document immutability of collections returned
The Features API returns collections in many cases.
These collections should be immutable and this should be documented.

### PR DESCRIPTION
**Pull request by bosschaert**
Document the unmodifiability of returned collections and potential
exceptions thrown.

Fixes #291

Signed-off-by: David Bosschaert <bosschae@adobe.com>

### CODE REVIEW

File: org.osgi.service.feature/src/org/osgi/service/feature/FeatureExtension.java [hunk 1]
```diff
@@ -97,22 +97,29 @@
...
-     * @return The JSON, or {@code null} if this is not a JSON extension.
-     */
-    String getJSON();
+	 * Get the JSON from this extension.
+	 * 
+	 * @return The JSON.
+	 * @throws IllegalStateException If called on an extension which is not of
+	 *             type JSON.
+	 */
+	String getJSON() throws IllegalStateException;
```
Comment (bjhargrave) [inline org.osgi.service.feature/src/org/osgi/service/feature/FeatureExtension.java:106]: In the OSGi API, we do not declare unchecked exceptions in the throws clause. Just in the javadoc.

File: org.osgi.service.feature/src/org/osgi/service/feature/FeatureExtension.java [hunk 2]
```diff
@@ -97,22 +97,29 @@
...
 	 * Get the Text from this extension.
 	 * 
-	 * @return The Text line by line, or {@code null} if this is not a Text
-	 *         extension.
+	 * @return The lines of text. The returned list is unmodifiable.
+	 * @throws IllegalStateException If called on an extension which is not of
+	 *             type TEXT.
 	 */
-	List<String> getText();
+	List<String> getText() throws IllegalStateException;
```
Comment (bjhargrave) [inline org.osgi.service.feature/src/org/osgi/service/feature/FeatureExtension.java:115]: In the OSGi API, we do not declare unchecked exceptions in the throws clause. Just in the javadoc.

File: org.osgi.service.feature/src/org/osgi/service/feature/FeatureExtension.java [hunk 3]
```diff
@@ -97,22 +97,29 @@
...
-     * @return The Artifacts, or {@code null} if this is not an Artifacts extension.
-     */
-    List<FeatureArtifact> getArtifacts();
+	 * Get the Artifacts from this extension.
+	 * 
+	 * @return The Artifacts. The returned list is unmodifiable.
+	 * @throws IllegalStateException If called on an extension which is not of
+	 *             type ARTIFACTS.
+	 */
+	List<FeatureArtifact> getArtifacts() throws IllegalStateException;
```
Comment (bjhargrave) [inline org.osgi.service.feature/src/org/osgi/service/feature/FeatureExtension.java:124]: In the OSGi API, we do not declare unchecked exceptions in the throws clause. Just in the javadoc.

**Comment by stbischof** [comment]
lgtm!

**Review by bjhargrave — APPROVED** [review]

**Pull request merged**